src/transformers/cache_utils.py

```diff
@@ -1428,11 +1428,7 @@ def from_legacy_cache(
     def get_seq_length(self, layer_idx: Optional[int] = 0) -> int:
         """Returns the sequence length of the cached states. A layer index can be optionally passed."""
         # check if empty list because in case of static cache it will be a tensors and we can't check `if not torch.Tensor`
-        if self.self_attention_cache.key_cache == []:
-            return 0
-        if len(self.self_attention_cache.key_cache) > 1 and self.self_attention_cache.key_cache[layer_idx] == []:
-            return 0
-        return (self.self_attention_cache.key_cache[layer_idx][0, 0].any(dim=-1)).sum()
+        return self.self_attention_cache.get_seq_length(layer_idx)
```

Comment:
got an error in one of the compiles when `get_seq_length()`, apparently we can't check if `tensor == []` when compiling. I think we can safely pass it to the self-attn cache

Reply:
you should be able to check the length ! ([] lives on cpu which is probably the source of the issue) having tensors for everything might be better btw

Reply:
yep, we should not go for that check on compiled model. That one is only for dynamic cache BC with empty lists